app/infrastructure/services/password_reset_service.py

```
@@ -0,0 +1,57 @@
+
+        Returns:
+            str: the email associated with the token if matched and deleted
+            None: when token is missing/expired
+        """
+        key = self._token_key(token)
+        redis = await get_redis()
+        try:
+            result = await redis.eval(self._CONSUME_LUA, 1, key)
+            if isinstance(result, bytes):
+                return result.decode("utf-8")
+            return result
```

Ruff (0.15.0)

[warning] Consider moving this statement to an `else` block

(TRY300)